scripts/docs-collator/ComponentRenderer.py

```diff
@@ -43,11 +51,12 @@ def __render_doc(self, component, file):
         relative_path = os.path.relpath(file, module_download_dir)
         result_file = os.path.join(self.docs_dir, os.path.relpath(file, module_download_dir))  # <module-name>/README.md
 
-        label = component
-        title = component
+        name = component if os.path.basename(file) == "README.md" else os.path.basename(file).replace(".md", "")
+        label = name
+        title = name
         github_edit_url = f"https://github.com/{GITHUB_REPO}/blob/master/modules/{relative_path}"
 
-        if len(relative_path.split('/')) > 2:  # this is submodule
+        if len(relative_path.split('/')) > 2 and relative_path.split('/')[1] != 'docs' :  # this is submodule
```

Comment:
This smells. Though I'm not sure how we want to handle this, we need to enforce either `docs/` or `modules` but think of tgw which uses `tgw/hub` as a component, it is nested, therefore I stuck with `docs` as an exception to the rule (where it's not a directory of TF)